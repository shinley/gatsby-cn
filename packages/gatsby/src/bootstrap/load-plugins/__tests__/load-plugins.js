const loadPlugins = require(`../index`)

describe(`Load plugins`, () => {
  it(`Load plugins for a site`, async () => {
    let plugins = await loadPlugins({ plugins: [] })

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })

  it(`Loads plugins defined with an object but without an option key`, async () => {
    const config = {
      plugins: [
        {
          resolve: `___TEST___`,
        },
      ],
    }

    let plugins = await loadPlugins(config)

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })
})
