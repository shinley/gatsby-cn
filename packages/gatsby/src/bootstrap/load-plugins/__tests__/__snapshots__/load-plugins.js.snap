// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load plugins Load plugins for a site 1`] = `
Array [
  Object {
    "browserAPIs": Array [],
    "id": "Plugin dev-404-page",
    "name": "dev-404-page",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin component-page-creator",
    "name": "component-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin component-layout-creator",
    "name": "component-layout-creator",
    "nodeAPIs": Array [
      "createLayouts",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin internal-data-bridge",
    "name": "internal-data-bridge",
    "nodeAPIs": Array [
      "sourceNodes",
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin prod-404",
    "name": "prod-404",
    "nodeAPIs": Array [
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin query-runner",
    "name": "query-runner",
    "nodeAPIs": Array [
      "onCreatePage",
      "onCreateLayout",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin default-site-plugin",
    "name": "default-site-plugin",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "d41d8cd98f00b204e9800998ecf8427e",
  },
]
`;

exports[`Load plugins Loads plugins defined with an object but without an option key 1`] = `
Array [
  Object {
    "browserAPIs": Array [],
    "id": "Plugin dev-404-page",
    "name": "dev-404-page",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin component-page-creator",
    "name": "component-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin component-layout-creator",
    "name": "component-layout-creator",
    "nodeAPIs": Array [
      "createLayouts",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin internal-data-bridge",
    "name": "internal-data-bridge",
    "nodeAPIs": Array [
      "sourceNodes",
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin prod-404",
    "name": "prod-404",
    "nodeAPIs": Array [
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin query-runner",
    "name": "query-runner",
    "nodeAPIs": Array [
      "onCreatePage",
      "onCreateLayout",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "name": "TEST",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
  },
  Object {
    "browserAPIs": Array [],
    "id": "Plugin default-site-plugin",
    "name": "default-site-plugin",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "d41d8cd98f00b204e9800998ecf8427e",
  },
]
`;
